core/com.b2international.snowowl.datastore/src/com/b2international/snowowl/datastore/request/version/CodeSystemVersionCreateRequest.java

```diff
@@ -124,18 +124,58 @@ public Boolean execute(ServiceProvider context) {
 		} catch (NotFoundException e) {
 			// ignore
 		}
+
+		final Map<CodeSystemEntry, IVersioningManager> versioningManagersByCodeSystem = createVersioningManagers(context, codeSystem);
+		for (Entry<CodeSystemEntry, IVersioningManager> entry : versioningManagersByCodeSystem.entrySet()) {
+			createVersion(context, entry.getValue(), entry.getKey(), user);
+		}
 		
-		// check that the specified effective time
-		validateEffectiveTime(context, codeSystem);
+		return Boolean.TRUE;
+	}
+	
+	private Map<CodeSystemEntry, IVersioningManager> createVersioningManagers(final ServiceProvider context, final CodeSystemEntry codeSystem) {
+		final Map<CodeSystemEntry, IVersioningManager> versioningManagersByCodeSystem  = Maps.newHashMap();
+		
+		final IVersioningManager versioningManager = VersioningManagerBroker.INSTANCE.createVersioningManager(codeSystem.getTerminologyComponentId());
+		versioningManagersByCodeSystem.put(codeSystem, versioningManager);
+
+		if (versioningManager.needsOtherCodeSystemsToVersion()) {
+			final Map<String, String> codeSystemsByRepositoryId = versioningManager.codeSystemDependenciesByRepositoryId();
+			for (Entry<String, String> entry : codeSystemsByRepositoryId.entrySet()) {
```

Comment:
Also this entire logic could go to the `versionManager`, we simply ask the "root" `versionManager` to gather and return all dependent version managers along with itself. Other code that depends on this logic can reuse that functionality via the well-named method.